### PR TITLE
Refactor Guardian contract and optimize PNS Schema

### PR DESCRIPTION
**Pull request by njokuScript**

### CODE REVIEW

**Review by bizmindx — CHANGES_REQUESTED** [review]
Left changes

File: contracts/Interfaces/IPNSSchema.sol [hunk 1]
```diff
@@ -21,9 +21,14 @@ interface IPNSSchema {
 		bool exists;
 		bool isInGracePeriod;
```
[the remark this answers is not in the document]
Comment (njokuScript) [reply contracts/Interfaces/IPNSSchema.sol:22]: It's being used in a couple of places not just to check but necessary to update the state of the phone if it has expired and puts it in grace period

File: contracts/Interfaces/IPNSSchema.sol [hunk 2]
```diff
@@ -21,9 +21,14 @@ interface IPNSSchema {
...
 		bool isExpired;
-		bool isVerified;
 		uint256 expirationTime;
-		uint256 verifiedAt;
 		uint256 createdAt;
```
[the remark this answers is not in the document]
Comment (MeisterJustice) [reply contracts/Interfaces/IPNSSchema.sol:25]: I don't think it's actually irrelevant. It's best to keep track of this on our end with ease.

**Review by MeisterJustice — APPROVED** [review]
Looks good to me

**Review by bizmindx — APPROVED** [review]
Well since we are quick on time  ✅

File: contracts/Interfaces/IPNSSchema.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (bizmindx) [reply contracts/Interfaces/IPNSSchema.sol:22]: this can be a method and extended into those checks, as opposed to it being a storage state.

**Pull request merged**